src/relay.ts

```diff
@@ -66,32 +66,59 @@ export default async function relay(
     logger.info(`Relay succeeded: ${hash}`);
     return true;
   } catch (err) {
-    if (err instanceof BaseError) {
-      const revertError = err.walk(
-        (err) => err instanceof ContractFunctionRevertedError,
-      );
-      if (revertError instanceof ContractFunctionRevertedError) {
-        const errName = revertError.data?.errorName ?? "";
-        if (errName !== "Error") {
-          // One of the custom errors we defined, which should include
-          // a more detailed message in the metaMessages field
-          const metaMsg = revertError.metaMessages
-            ?.map((msg: string) => msg.trim())
-            .join("");
+    if (!(err instanceof BaseError)) {
+      // Theoretically should never happen, as all errors in Viem should extend BaseError
+      logger.error("Relay failed due to an unknown non-BaseError:", err);
+      return false;
+    }
 
-          logger.error("Contract reverted with error:", metaMsg);
-        } else {
-          // A generic revert error, which should include the reason
-          logger.error("Contract reverted with reason:", revertError.reason);
+    const revertError = err.walk(
+      (err) => err instanceof ContractFunctionRevertedError,
+    );
+    if (revertError instanceof ContractFunctionRevertedError) {
```

Comment:
is this necessary for typescript?

line 76 looks as if we're already filtering out only `ContractFunctionRevertedError`s there so this `if` feels redundant

Reply:
It is because err.walk() can return null in case the error is not of type `ContractFunctionRevertedError`

Reply:
ah i see, thx for explaining